check-github-issue.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+
+import argparse
+import requests
+import os
+import json
+import subprocess
+import re
+from github3 import login
+
+def get_commit_message():
+    commit = subprocess.check_output(['git', 'log', '--format=%B', '-n', '1'])
+    return commit
+
+
+def get_commit_message_from_gerrit(project, branch, change_id, revision_number):
+    k = requests.get('https://review.gluster.org/changes/{}~{}~{}'
+                     '/revisions/{}/commit'.
+                     format(
+                            project,
+                            branch,
+                            change_id,
+                            revision_number
+                            )
+                     )
+    output = k.text
+    cleaned_output = '\n'.join(output.split('\n')[1:])
+    parsed_output = json.loads(cleaned_output)
+    # TODO: check if we got valid data before returning the same?
+    return parsed_output.get("message")
+
+
+def parse_commit_message(msg):
+    regex = re.compile(r'(([fF][iI][xX][eE][sS])|([uU][pP][dD][aA][tT][eE][sS])):?\s+(gluster/glusterfs)?#(\d*)')
+    issues = []
+    for line in msg.split('\n'):
+        for match in regex.finditer(line):
+            issue=match.group(5)
+            if issue.to_i >= 743000:
+                continue
+            issues.append(unicode(issue))
+    if len(issues):
+        print("Issues found in the commit message: {}".format(issues))
+        return issues
+    print("No issues found in the commit message")
+    return issues
+
+
+def remove_duplicates (project, branch, change_id, revision_number, issues):
+    if (int(revision_number) == 1):
+        return issues
+
+    oldmessage = get_commit_message_from_gerrit(project, branch, change_id,
+                                                str(int(revision_number) - 1))
+    oldissues = parse_commit_message(oldmessage)
+    print("Old issues: {}".format(oldissues))
+    newissues = list(set(issues) - (set(issues) & set(oldissues)))
+    return newissues
+
+def check_issue(num, comment, dry_run):
+    if dry_run:
+        print comment
+    else:
+        gh_user = os.environ.get('GITHUB_USER')
+        gh_pw = os.environ.get('GITHUB_PASS')
+        gh = login(gh_user, gh_pw)
+        issue = gh.issue('gluster', 'glusterfs', num)
+        if issue:
+            spec_approved = False
+            doc_approved = False
+            for lbl in issue.labels:
+                if lbl == "SpecApproved":
+                    spec_approved = True
+                if lbl == "DocApproved":
+                    doc_approved = True
+            if spec_approved and doc_approved:
+                print "All approvals in place"
+                exit (0)
+            else:
+                print "Missing the required approvals"
```

Comment:
I assume this would be part of the smoke job, and so it would be nice if we can have a more descriptive failure, and message to recover from that failure (IOW, which issue failed, and due to which flags, and also what to do next).

Reply:
Ack!